Net011/TrainingLesson.cs

```diff
@@ -2,23 +2,83 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Text;
-using System.Threading.Tasks;
 
 namespace Net011
 {
-    public class TrainingLesson : TrainingEntity
+    public class TrainingLesson : TrainingEntity, IVersionable, ICloneable
     {
-        private string _trainingLessonType;
+        private byte[] _version = new byte[8];
         public List<TrainingMaterial> TrainingMaterials { get; set; } = new List<TrainingMaterial>();
         public TrainingLesson()
         {
         }
 
-        // includes a set of training materials
-        // for example: trainingLessonOne = 5x text material, 2x video material, 1x link to a network resource
-        public string GetTrainingType()
+        public LessonType GetTrainingType()
         {
-            return string.Empty;
+            //var res1 = TrainingMaterials.Where(x => x.Description == "description one");            // kol nepanaudota, ToList, uzklauso nevykdoma/negauna rezultato
+            //var res2 = TrainingMaterials.Where(x => x.Description == "description one").ToList();   // panaudojus ToList, uzklausa ivykdyta/gautas rezultatas
+            ////var res3 = TrainingMaterials.Where(x => x.Description == "description two").First();
+
+            //try
+            //{
+            //    var res3 = TrainingMaterials.Where(x => x.Description == "description two").First();
+            //}
+            //catch (System.Exception e)
+            //{
+            //    System.Console.WriteLine(e.Message);
+            //    throw;
+            //}
+
+            //var res4 = TrainingMaterials.Where(x => x.Description == "description two").FirstOrDefault();
+            if (TrainingMaterials.Any(x => x.GetType() == typeof(VideoMaterial)))
+            {
+                return LessonType.VideoLesson;
+            }
+            else
+            {
+                return LessonType.TextLesson;
+            }
+        }
+
+        public override string ToString()
+        {
+            return Description;
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj == null)
+            {
+                return false;
+            }
+            if (!(obj is TrainingMaterial))
+            {
+                return false;
+            }
+            return (Id == ((TrainingMaterial)obj).Id);
+        }
+
```

Comment:
Why 0? )))

Reply:
This line is added automatically by the compiler so that I won't get a "warning" message in the Error list :D I have no idea what is the real solution for the GetHashCode() method here.